packages/backend/src/core/queue/promiseAllPlus.ts

```diff
@@ -49,6 +50,7 @@ export async function promiseAllPlus<T>(
       }
       return shouldRetry
     },
+    metricsId: opts?.metricsId ?? 'undefined',
```

Comment:
I don't think this is a good approach. Either this id should be required or not.